src/app/auth.guard.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
+import { Observable } from 'rxjs/Observable';
+import { UserService } from './services/user.service';
+
+@Injectable()
+export class AuthGuard implements CanActivate {
+
+  constructor(private UserService: UserService) {
+  }
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
+      if ( this.UserService.getUser()) {
```

Comment:
Could be simplified to:

```
return this.UserService.getUser();
```

also, when naming variable / properties of a class, make sure to use `camelCase`. 
I.e. `this.UserService` should actually be `this.userService.getUser()`

Reply:
oops....